w3-store.md

```diff
@@ -1,631 +1,870 @@
-| `origin` | CAR CID string, e.g. `bagabc...`  |             | Optional link to related CARs. See below for more details. |
+##### Store Add Origin
 
-#### Invocation
+> Status: Deprecated 🛑
+>
+> Field was intended for establish causal order, but proved impractical.
 
-To invoke `store/add`, an agent constructs a UCAN with the shape described below.
+Capability invocation MAY set _optional_ `nb.origin` field to a causally related [CAR] like a previous shard of the content DAG.
 
-Example:
+##### Store Add Example
 
 ```js
 {
-  can: "store/add",
-  with: "did:key:abc...",
-  nb: {
-    link: "bag...",
-    size: 1234
+  "v": "0.9.1",
+  "iss": "did:key:z6Mkk...xALi",
+  "aud": "did:web:web3.storage",
+  "att": [
+    {
+      can: "store/add",
+      with: "did:key:zAl..1ce",
+      nb: {
+        link: { "/": "bag...7ldq" },
+        size: 42_600_000
+      }
+    }
+  ],
+  "prf": [{ "/": "bafy...prf1" }],
+  "exp": 1685602800,
+  "s": {
+    "/": {
+      "bytes": "7aEDQJbJqmyMxTxcK05XQKWfvxG+Tv+LWCJeE18RSMnciCZ/RQ21U75LA0uFSvIjdqnF5RaauZTE8mh2ZYMBBejdJQ4"
+    }
   }
 }
 ```
 
-The `nb.origin` field may be set to provide a link to a related CAR file. This is useful when storing large DAGs that are sharded across multiple CAR files. In this case, the agent can link each uploaded shard with a previous one. Providing the `origin` field informs the service that the CAR being stored is a shard of the larger DAG, as opposed to an intentionally partial DAG.
+#### Store Add Receipt
 
-Note that the `nb.link` and `nb.origin` caveats contain the CID of the CAR itself, not the CID of any content contained within the CAR. It should have the [multicodec code value `0x0202`](https://github.com/multiformats/multicodec/blob/master/table.csv#L135), which is reserved for data in CAR format. When encoded to a CID string using the default encoding, this results in a CID with a prefix of `bagb`.
+Provider MUST issue signed receipt containing `StoreAddResult`.
 
-#### Responses
+##### Store Add Result
 
-*Note*: This section is non-normative and subject to change, pending the [formalization of receipts and invocations][invocation-spec-pr].
+###### Store Add Result IPLD Schema
 
-Executing a `store/add` invocation with a service provider should return a result object.
+```ipldsch
+type StoreAddResult union {
+  StoreAddSuccess   "ok"
+  StoreAddFailure   "error"
+} representation keyed
 
-If the invocation fails, the result will include an `error` field with a value of `true`, and a `message` field containing additional details.
+type StoreAddSuccess union {
+  StoreAddDone      "done"
+  StoreAddPending   "upload"
+} representation inline {
+  discriminantKey "status"
+}
 
-On success, the response will include a `status` field, which may have one of the following values:
+type StoreAddDone struct {
+  with    SpaceDID
+  link    &ContentArchive
```

Comment:
we miss allocated property (this is used for billing to get to know how much to bill to the space). see:
- https://github.com/web3-storage/w3up/blob/main/packages/upload-api/src/store/add.js#L67
- https://github.com/web3-storage/w3up/pull/1213